### ISSUE TITLE
API keys not auto-generated, even with AUTOGENERATE_API_KEY = True in settings.py

### ISSUE THREAD

**Issue by wettenhj** [mytardis/mytardis#381]
This suggests a problem with post-save hooks for user records, as does: https://github.com/mytardis/mytardis/issues/379

API keys get generated fine when I run "bin/django backfill_api_keys", but I shouldn't need to do that if I have "AUTOGENERATE_API_KEY = True" in settings.py

**Comment by wettenhj** [comment]
The code to connect the post_save signal for User records to tastypie.models.create_api_key is here: https://github.com/mytardis/mytardis/blob/develop/tardis/tardis_portal/models/hooks.py#L69, but it isn't being run.  All we need to do to make this code run is to add "```import tardis.tardis_portal.models.hooks```" somewhere - perhaps in tasks.py, near where we initialize the DataFile post_save hooks using FilterInitMiddleware: https://github.com/mytardis/mytardis/blob/develop/tardis/tardis_portal/tasks.py#L19

Individual functions from hooks.py are already being imported, e.g. in tardis_portal/models/experiment.py, we have:

```from .hooks import publish_public_expt_rifcs```

but that wont' load the module-level code in tardis/tardis_portal/models/hooks.py needed to autogenerate the API keys.

**Comment by wettenhj** [comment]
Also, post_save.connect seems to be more reliable with weak=False, as noted in tardis/tardis_portal/filters/\_\_init\_\_.py, i.e.
```
--- a/tardis/tardis_portal/models/hooks.py
+++ b/tardis/tardis_portal/models/hooks.py
@@ -70,8 +70,9 @@ def ensure_doi_exists(sender, **kwargs):
 if getattr(settings, 'AUTOGENERATE_API_KEY', False):
     from django.contrib.auth.models import User
     from tastypie.models import create_api_key
-    post_save.connect(create_api_key, sender=User)
-
+    post_save.connect(create_api_key, sender=User, weak=False)
+    logger.info('Initialised postsave hooks for ApiKeys: %s'
+                % post_save.receivers)
```

**Comment by grischa** [comment]
closed by commit a11460327e455cfb29200183324623a4eb7b36a7
